src/realm_dart_sync.cpp

```diff
@@ -139,56 +139,46 @@ RLM_API void realm_dart_sync_on_subscription_state_changed_callback(realm_userda
     });
 }
 
-bool wait_dart_before_continue(realm::util::UniqueFunction<void(realm::util::UniqueFunction<void(bool)>*)>* userCallback)
+bool invoke_dart_and_await_result(realm::util::UniqueFunction<void(realm::util::UniqueFunction<void(bool)>*)>* userCallback)
```

Comment:
I prefer more `wait_dart_before_continue` since when you see the place where it is called it is clear that you stops the execution there and you wait something. Could be `wait_dart_result_then_continue`. But starting with `wait` makes it more obvious that you are waiting.

Reply:
There is a `wait` in the function name. Its just more natural language. The previous function name was not natural to me.